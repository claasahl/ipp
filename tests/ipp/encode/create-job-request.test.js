const encode = require("../../../build/ipp/encode").default;

test("Create-Job Request", () => {
  const message = {
    type: "IppRequest",
    versionNumber: { major: 0x01, minor: 0x01},
    operationId: 0x0005,
    requestId: 0x00000001,
    attributeGroup: [
        {
            type: "AttributeGroup",
            beginAttributeGroupTag: 0x01,
            attribute: [
                {
                    type: "Attribute",
                    attributeWithOneValue: {
                        type: "AttributeWithOneValue",
                        valueTag: 0x47,
                        nameLength: 0x0012,
                        name: "attributes-charset",
                        valueLength: 0x0005,
                        value: "utf-8",
                    },
                    additionalValue: []
                },
                {
                    type: "Attribute",
                    attributeWithOneValue: {
                        type: "AttributeWithOneValue",
                        valueTag: 0x48,
                        nameLength: 0x001b,
                        name: "attributes-natural-language",
                        valueLength: 0x0005,
                        value: "en-us",
                    },
                    additionalValue: []
                },
                {
                    type: "Attribute",
                    attributeWithOneValue: {
                        type: "AttributeWithOneValue",
                        valueTag: 0x45,
                        nameLength: 0x000b,
                        name: "printer-uri",
                        valueLength: 0x002c,
                        value: "ipp://printer.example.com/ipp/print/pinetree",
                    },
                    additionalValue: []
                },
            ]
        }
    ],
    endOfAttributesTag: 0x03,
    data: Buffer.from([])
}

  const data = encode(message);
  expect(data.toString("hex")).toMatchSnapshot();
});
