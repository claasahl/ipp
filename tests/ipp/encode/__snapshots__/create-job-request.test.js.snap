// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create-Job Request 1`] = `"010100050000000101470012617474726962757465732d6368617273657400057574662d3848001b617474726962757465732d6e61747572616c2d6c616e67756167650005656e2d757345000b7072696e7465722d757269002c6970703a2f2f7072696e7465722e6578616d706c652e636f6d2f6970702f7072696e742f70696e657472656503"`;
